runtime/nvqir/QIRTypes.cpp

```diff
@@ -19,6 +19,42 @@ extern bool initialized;
 extern bool verbose;
 }
 
+namespace {
+// Singleton struct to track allocated arrays
+struct ArrayTracker {
+  // Get the singleton instance of ArrayTracker
+  static ArrayTracker &getInstance() {
+    static ArrayTracker instance;
```

Comment:
I suppose we could use the C++ run once stuff to avoid the static initializer here. Kind of nitpicky though.

Reply:
Yeah, I think I'll stick with Meyer’s singleton pattern here for its simplicity :)